js/src/setup-ads/ads-stepper/index.js

```diff
@@ -12,21 +12,27 @@ import SetupAccounts from './setup-accounts';
 import AppButton from '.~/components/app-button';
 import AdsCampaign from '.~/components/paid-ads/ads-campaign';
 import useEventPropertiesFilter from '.~/hooks/useEventPropertiesFilter';
+import useGoogleAdsAccountBillingStatus from '.~/hooks/useGoogleAdsAccountBillingStatus';
+import { GOOGLE_ADS_BILLING_STATUS } from '.~/constants';
 import {
 	recordStepperChangeEvent,
 	recordStepContinueEvent,
 	FILTER_ONBOARDING,
 	CONTEXT_ADS_ONBOARDING,
 } from '.~/utils/tracks';
 
+const { APPROVED } = GOOGLE_ADS_BILLING_STATUS;
+
 /**
  * @param {Object} props React props
- * @param {Object} props.formProps Form props forwarded from `Form` component.
+ * @param {Object} props.isSubmitting When the form in the parent component, i.e SetupAdsForm, is currently being submitted via the useAdsSetupCompleteCallback hook.
  * @fires gla_setup_ads with `{ triggered_by: 'step1-continue-button', action: 'go-to-step2' }`.
  * @fires gla_setup_ads with `{ triggered_by: 'stepper-step1-button', action: 'go-to-step1'}`.
  */
-const AdsStepper = ( { formProps } ) => {
+const AdsStepper = ( { isSubmitting } ) => {
 	const [ step, setStep ] = useState( '1' );
+	const { billingStatus } = useGoogleAdsAccountBillingStatus();
```

Comment:
Without a connected Google Ads account, entering the Ads Setup flow will cause critical errors.

![image](https://github.com/user-attachments/assets/f29fda32-bb40-4215-b564-c1e00cc603b8)

If the reason for making the continue button a standalone component (`LaunchPaidCampaignButton`) was to avoid these errors, then I would suggest using a similar structure as `SetupPaidAds` to isolate the following scope.

This would create a more consistent level of abstraction within `AdsStepper`, and the continue button implementation could still be placed together with `AdsCampaign` for better readability.

Reply:
Thanks for the suggestion @eason9487 . I've added another `SetupPaidAds` component where we now have `AdsCampaign` along with the billing status check. I've kept the `SetupPaidAds` name to be consistent.